app/serializers/chat_base_message_serializer.rb

```diff
@@ -11,13 +11,35 @@ class ChatBaseMessageSerializer < ApplicationSerializer
     :deleted_at,
     :deleted_by_id,
     :flag_count,
-    :edited
+    :edited,
+    :reactions
 
   has_one :user, serializer: BasicUserSerializer, embed: :objects
   has_one :chat_webhook_event, serializer: ChatWebhookEventSerializer, embed: :objects
   has_one :in_reply_to, serializer: ChatBaseMessageSerializer, embed: :objects
   has_many :uploads, serializer: UploadSerializer, embed: :objects
 
+  def reactions
+    reactions_hash = {}
+    object.reactions.group_by(&:emoji).each do |emoji, reactions|
+      users = reactions[0..6].map(&:user).filter { |user| user.id != scope.user.id }[0..5]
+      reactions_hash[emoji] = {
+        count: reactions.count,
+        users: ActiveModel::ArraySerializer.new(users, each_serializer: BasicUserSerializer).as_json,
+        reacted: users_reactions.include?(emoji)
+      }
+    end
+    reactions_hash
+  end
+
+  def include_reactions?
+    object.reactions.any?
```

Comment:
Could you double check this doesn't cause an extra `SELECT 1 FROM…` query?

Reply:
I did, yes!